lasagne/utils.py

```diff
@@ -76,6 +76,29 @@ def as_theano_expression(input):
                             "exception: %s)" % (type(input), e))
 
 
+def collect_shared_vars(expressions):
+    """Returns all shared variables the given expression(s) depend on.
+
+    Parameters
+    ----------
+    expressions : Theano expression or iterable of Theano expressions
+        The expressions to collect shared variables from.
+
+    Returns
+    -------
+    list of Theano shared variables
+        All shared variables the given expression(s) depend on, in the order
+        found by a left-recursive depth-first search. If some expressions are
```

Comment:
Is there any particular reason you went with this search strategy? Just asking because it's mentioned explicitly. I guess it's just the easiest to implement?